### PR TITLE
🧹 switch to github.com/go-jose/go-jose/v3

### PR DESCRIPTION
**Pull request by chris-rock**
The square library is not supported anymore. Therefore we switch to the latest available one.

### CODE REVIEW

File: plugins/rangerguard/user/claim_test.go [hunk 1]
```diff
@@ -8,10 +8,10 @@ import (
 	"testing"
 	"time"
 
+	"github.com/go-jose/go-jose/jwt"
```
Comment (czunker) [inline plugins/rangerguard/user/claim_test.go:11]: `v3` is missing here.

File: go.mod [hunk 2]
```diff
@@ -4,6 +4,8 @@ go 1.21
 
 require (
 	github.com/cockroachdb/errors v1.11.1
+	github.com/go-jose/go-jose v2.6.1+incompatible
```
Comment (czunker) [inline go.mod:7]: I assume the change to `plugins/rangerguard/user/claim_test.go` removes this.

File: plugins/rangerguard/user/claim_test.go (hunk 1 above)
Comment (chris-rock) [reply plugins/rangerguard/user/claim_test.go]: great catch, PR updated

File: go.mod (hunk 2 above)
Comment (chris-rock) [reply go.mod]: indeed

**Pull request merged**